lib/spack/spack/test/cmd/external.py

```diff
@@ -240,3 +244,71 @@ def test_new_entries_are_reported_correctly(
     # has been found
     output = external('find', 'gcc')
     assert 'No new external packages detected' in output
+
+
+def candidate_packages():
+    all_packages = spack.repo.path.all_packages()
+    to_be_tested = []
+    for pkg in all_packages:
+        pkg_dir = os.path.dirname(
+            spack.repo.path.filename_for_package_name(pkg.name)
+        )
+        detection_data = os.path.join(pkg_dir, 'detection_test.yaml')
+        if os.path.exists(detection_data):
+            to_be_tested.append(pkg.name)
+    return to_be_tested
```

Comment:
Note that this will slow-down start-up of unit tests, but at the moment I can't think of a better way to retrieve this information.

Reply:
This is a function to initialize a fixture, so it's run when tests are collected not when they are run - which is not optimal. I'll take your suggestion of not going through the import mechanism and see if that improves the current situation.

Reply:
Implemented your suggestion and used `glob.glob`. Brilliant, it's so much faster! :tada: